app/javascript/src/components/Dashboard/Contacts/Table.jsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+
+import { MenuHorizontal } from "neetoicons";
+import { Table as NeetoUITable, Avatar, Typography, Dropdown } from "neetoui";
+
+import { monthDateFormatter } from "utils/date";
+
+import { PROFILE_PICTURE_URL } from "./constants";
+
+const renderAvatarWithNameAndRole = (firstName, { lastName, role }) => (
+  <div className="flex space-x-2">
+    <Avatar size="large" user={{ imageUrl: PROFILE_PICTURE_URL }} />
+    <div className="flex flex-col">
+      <Typography style="h4">
+        {firstName} {lastName}
+      </Typography>
+      <Typography style="body3">{role}</Typography>
+    </div>
+  </div>
+);
+
+const renderOptionButton = () => (
+  <Dropdown buttonStyle="text" icon={MenuHorizontal}>
+    <div className="p-1.5">
+      <li>Edit</li>
+      <li>Delete</li>
+    </div>
+  </Dropdown>
+);
```

Comment:
@anandhakrishnanaji 
This is good abstraction. Can you please create a video and put it up in neeto-ui-challenge channel? That will help others. The video should cover these points:
- When do we need to make the judgement to move the JSX rendering logic into a separate function?
- What is the difference between writing this as a function vs writing it as a component? When to prefer which? We need to name it as a component when we need to make it a standalone component.
- Naming conventions for `render*` functions.
- We should also mention that we should pass in the arguments required for the working of the `render*` functions rather than taking them from the scope. The reason is that, then only we can easily move out these `render` functions into some other file when the current component becomes too big. You will have to first make related changes in your code before talking about this point.

Reply:
Sure 👍

Reply:
PS: You should make the following changes:
- Rename `renderOptionButton` to `renderActionDropdown`. You are not simply rendering a single Button.
- Make the whole arguments of `renderAvatarWithNameAndRole` named arguments. Right now you have made the first argument a positional argument and the rest as named arguments, which is not right.

Reply:
```suggestion
const renderAvatarWithNameAndRole = (_, {firstName, lastName, role }) => (
```
You meant like this right?
I am using the function with the Table component and it passes the key used as the first positional argument.

Reply:
@anandhakrishnanaji Ok. In that case we can stick onto the current argument format itself.